libglusterfs/src/glusterfs/mem-pool.h

```diff
@@ -21,6 +21,12 @@
 #include <string.h>
 #include <stdarg.h>
 
+#include <sys/param.h> /* roundup */
+#ifndef roundup
+/* not a standard, so some platforms may not have it. Define a generic one */
+# define roundup(x, y)  ((((x) + ((y) - 1)) / (y)) * (y))
```

Comment:
Our allocation speed isn't the greatest (using tcmalloc and disabling memory pools seem to provide better performance) - are we sure we want to introduce division and multiplication to the allocation path?

Reply:
I'd be happy to update this patch to just remove the trailer magic and complexity associated. Faster in the general case, but does not preclude using other tools (runtime or compile) for debugging malloc.

Reply:
I think this is a question for a broader forum of the developer community than in a PR - can you send such a question to the devel mailing list?

Reply:
Worth noting the glibc roundup uses a GCC builtin, which IIRC is a single instruction on x86. Other platforms might have similar.

I've added this one for platform compatibility.

Reply:
Given that `y` should always be a power of 2, why not this:

```c
#define roundup(_x, _y) (((_x) + (_y) - 1) & ~((_y) - 1))
```